src/locale/keymap-util.c

```diff
@@ -780,3 +782,150 @@ int x11_convert_to_vconsole(Context *c) {
 
         return modified;
 }
+
+static bool locale_encoding_is_utf8_or_unspecified(const char *locale) {
+        const char *c = strchr(locale, '.');
+        return !c || strcaseeq(c,  ".UTF-8") || strcasestr(locale, ".UTF-8@");
+}
+
+bool locale_gen_check_available(void) {
+#if HAVE_LOCALEGEN
+        if (access(LOCALEGEN_PATH, X_OK) < 0) {
+                if (errno != ENOENT)
+                        log_warning_errno(errno, "Unable to determine whether " LOCALEGEN_PATH " exists and is executable: %m");
+                return false;
+        }
+        if (access("/etc/locale.gen", F_OK) < 0) {
+                if (errno != ENOENT)
+                        log_warning_errno(errno, "Unable to determine whether /etc/locale.gen exists: %m");
+                return false;
+        }
+        return true;
+#else
+        return false;
+#endif
+}
+
+int locale_gen_enable_locale(char *locale[_VARIABLE_LC_MAX]) {
```

Comment:
wouldn't it make more sense to just accept a single string here, instead of the full array? i.e. just the locale to actually add?